### DISCUSSION TITLE
Structure output

### DISCUSSION THREAD

**Discussion by molecularmachinist** [Vattulainen-Group/CellSimGmx#1] [Ideas]
Hi! As we plan to model our framework for the tissue architecture after a protein (with particles ->atoms, Cells -> Residues and Layers ->Chains) I suggest we make PDB format structure output as default. GRO format should be discouraged or even made inadmissible as GRO does not allow for chain information in it. This way we will not lose Layer information in the structure and we will be able to read the PDB file directly to system objects without any ambiguity. 

@svihuri @kkorshunova

**Comment by kkorshunova** [comment]
@molecularmachinist while I do agree that PDB format would be more helpful for the "input preparation" and "analysis" modules, would that mean that we will need extra pdb2gro and gro2pdb tools in order to interface with GROMACS?

I suppose, one can use standard gmx tools for the first part (pdb2gro after input preparation).
However, for processing GROMACS output (the gro file) and converting it into pdb again, we would probably need a custom tool that should be able to correctly assign the chain labels, right?

**Comment by svihuri** [comment]
@molecularmachinist I'm not too familiar with the difference between gro and pdb files, what layer information would there be in the pdb file that we need for the system formation?

**Comment by molecularmachinist** [comment]
Hello. 

@kkorshunova To answer your question: We wont have to depend on gromacs at all. We are currently writing out the GRO file with a custom function that takes in the particles information as input. I have added another custom function that write PDB files instead with a dictionary input. @svihuri Can you please have a look in the branch write_pdb? I have assigned you and Ksenia as the code reviewers. Read the code and try it out. It will write out a PDB with two atoms to test. If you feel it is up to expectations then you can merge it to main branch. I have also created a directory structure that we can use. You can also suggest changes to it.

@svihuri 

The difference is cosmetic however the PDB file can store more information such as chainID which we will need to recover the system object from the structure file. GRO file format does not store the chainID so we can not use it to go back the system object. It is a sort of then a necessity to use the PDB format. The code I wrote has the PDB format in it. The dictionary in the signature provides an interface to talk to the PDB format. 

To wit: the chainID in the PDB stores the LayerID for us. Which we need to assign cells (ResID) to the correct layer.

Shreyas
